kolibri/plugins/management/assets/src/actions.js

```diff
@@ -69,110 +79,142 @@ function _managePageTitle(title) {
-    FacilityUserModel.attributes.roles[0].id : null;
-  const oldRole = FacilityUserModel.attributes.roles.length ?
-    FacilityUserModel.attributes.roles[0].kind : 'learner';
-
-  if (oldRole !== role) {
-  // the role changed
-    if (oldRole === 'learner') {
-    // role is admin or coach.
-      const rolePayload = {
-        user: id,
-        collection: FacilityUserModel.attributes.facility,
-        kind: role,
-      };
-      const RoleModel = RoleResource.createModel(rolePayload);
-      RoleModel.save(rolePayload).then((newRole) => {
-        FacilityUserModel.save(payload).then(responses => {
-          // force role change because if the role is the only changing attribute
-          // FacilityUserModel.save() will not send request to server.
-          responses.roles = [newRole];
-          store.dispatch('UPDATE_USERS', [responses]);
-        })
-        .catch(error => { coreActions.handleApiError(store, error); });
-      });
-    } else if (role !== 'learner') {
-    // oldRole is admin and role is coach or oldRole is coach and role is admin.
-      const OldRoleModel = RoleResource.getModel(oldRoldID);
-      OldRoleModel.delete().then(() => {
-      // create new role when old role is successfully deleted.
-        const rolePayload = {
-          user: id,
-          collection: FacilityUserModel.attributes.facility,
-          kind: role,
-        };
-        const RoleModel = RoleResource.createModel(rolePayload);
-        RoleModel.save(rolePayload).then((newRole) => {
-        // update the facilityUser when new role is successfully created.
-          FacilityUserModel.save(payload).then(responses => {
-            // force role change because if the role is the only changing attribute
-            // FacilityUserModel.save() will not send request to server.
-            responses.roles = [newRole];
-            store.dispatch('UPDATE_USERS', [responses]);
-          })
-          .catch(error => { coreActions.handleApiError(store, error); });
-        });
-      })
-      .catch(error => { coreActions.handleApiError(store, error); });
-    } else {
-    // role is learner and oldRole is admin or coach.
-      const OldRoleModel = RoleResource.getModel(oldRoldID);
-      OldRoleModel.delete().then(() => {
-        FacilityUserModel.save(payload).then(responses => {
-          // force role change because if the role is the only changing attribute
-          // FacilityUserModel.save() will not send request to server.
-          responses.roles = [];
-          store.dispatch('UPDATE_USERS', [responses]);
-        })
-        .catch(error => { coreActions.handleApiError(store, error); });
+function updateUser(store, stateUser) {
+  //payload needs username, fullname, and facility
+  const userID = stateUser.id;
+  const savedUserModel = FacilityUserResource.getModel(userID);
+  let savedUser = savedUserModel.attributes;
+  let roleAssigned = Promise.resolve(savedUserModel.attributes);
+  let changedValues = {};
```

Comment:
Note that `const` only applies to the top-level reference. Here you only change attributes of `changedValues`, but the container object stays the same.

----

What follows is my subjective opinion & experience:

More generally (not specific to these examples) it's good to default to using `const` because it encourages a functional/declarative style rather than imperative coding.

Two examples are the discussion above (https://github.com/learningequality/kolibri/pull/721#discussion_r92271869) and the filter logic below (https://github.com/learningequality/kolibri/pull/721#discussion-diff-92266163R156). Both of these use an imperative style, and tend to be a bit more brittle than functional/declarative logic because they rely on keeping track of internal state.

If you find yourself initializing a `let` variable and then updating the value in a loop or if statements, it's often worth exploring whether a more functional approach could be used.